pennylane/drawer/mpldrawer.py

```diff
@@ -293,12 +293,36 @@ def __init__(self, n_layers, n_wires, c_wires=0, wire_options=None, figsize=None
 
         if wire_options is None:
             wire_options = {}
+            self._wire_lines = [
+                plt.Line2D((-1, self.n_layers), (wire, wire), zorder=1, **wire_options)
+                for wire in range(self.n_wires)
+            ]
+        else:
+            # Separate default options from specific wire options
+            default_options = {k: v for k, v in wire_options.items() if isinstance(k, str)}
+            wire_specific_options = {k: v for k, v in wire_options.items() if isinstance(k, int)}
```

Comment:
I should have mentioned this earlier on in the examples I think.
Wire labels are allowed to be any hashable object, so in particular strings are allowed.
I would therefore suggest to make the distinction of whether something is a per-wire option based on whether the key is in the wire labels. 
Now, this class does not even receive the wires but only the number of wires via `n_wires`, which would require us to change the interface of the class. I'd like to avoid this.
So as a surrogate, we could check whether the _value_ is a dictionary, and infer that if it is one, the key-value pair must've been a per-wire option. I'm not sure whether there are any line properties that themselves would have dictionaries as values, but it seems like the lesser evil at this point.

I think we should refine this somewhat hacky way of doing things later on, but for now I'd say this is a fair tradeoff between feature addition and tiny edge case (in which a dictionary-valued kwarg would be interpreted as per-wire option although it is meant to be an all-wires option).

Reply:
Sorry for not seeing this one coming earlier!

Reply:
Thanks a lot for the comment, I have modified the necessary part into dictionary checks.